src/logic/community/compliance-validator.ts

```diff
@@ -0,0 +1,98 @@
+}
+
+export function createCommunityComplianceValidatorComponent(
+  components: Pick<AppComponents, 'aiCompliance' | 'featureFlags' | 'logs'>
+): ICommunityComplianceValidatorComponent {
+  const { aiCompliance, featureFlags, logs } = components
+  const logger = logs.getLogger('community-compliance-validator')
+
+  return {
+    async validateCommunityContent(request: {
+      name: string
+      description: string
+      thumbnailBuffer?: Buffer
+    }): Promise<void> {
+      if (!featureFlags.isEnabled(FeatureFlag.COMMUNITIES_AI_COMPLIANCE)) {
+        logger.info('Skipping AI compliance validation for communities because the feature flag is disabled')
+        return
+      }
+
+      const { name, description, thumbnailBuffer } = request
+      const startTime = Date.now()
+
+      try {
+        logger.info('Starting community content compliance validation', {
+          name,
+          descriptionLength: description.length,
+          hasThumbnail: String(!!thumbnailBuffer)
+        })
+
+        const validationResult = await aiCompliance.validateCommunityContent({
+          name,
+          description,
+          thumbnailBuffer
+        })
+
+        const duration = Date.now() - startTime
+
+        if (!validationResult.isCompliant) {
+          logger.warn('Community content compliance validation failed', {
+            name,
+            issues: validationResult.issues.join(', '),
+            warnings: validationResult.warnings.join(', '),
+            confidence: validationResult.confidence,
+            duration
+          })
+
+          throw new CommunityComplianceError(
+            `Community content violates Decentraland's Code of Ethics: ${validationResult.reasoning}`,
+            validationResult.issues,
+            validationResult.warnings,
+            validationResult.confidence
+          )
+        }
+
+        if (validationResult.warnings.length > 0) {
+          logger.info('Community content compliance validation passed with warnings', {
+            name,
+            warnings: validationResult.warnings.join(', '),
+            confidence: validationResult.confidence,
+            duration
+          })
+        } else {
+          logger.info('Community content compliance validation passed', {
+            name,
+            confidence: validationResult.confidence,
+            duration
+          })
+        }
+      } catch (error) {
+        const duration = Date.now() - startTime
+
+        if (error instanceof CommunityComplianceError) {
+          throw error
+        }
+
+        logger.error('Community content compliance validation error', {
+          name,
+          error: errorMessageOrDefault(error, 'Unknown error'),
+          duration
+        })
+
+        throw new CommunityComplianceError(
+          'Unable to validate community content compliance. Manual review required.',
```

Comment:
Nice, maybe we can use a boolean or a different error type to differentiate both cases (revision failed or revision could not be performed)